distributions/openhab/src/main/resources/bin/backup.ps1

```diff
@@ -0,0 +1,118 @@
+        [Parameter(ValueFromPipeline=$True)]
+        [string]$OHBackups,
+        [Parameter(ValueFromPipeline=$True)]
+        [string]$ZipFileOut
+    )
+
+    begin {}
+    process {
+
+        if (!([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()
+            ).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
+            throw "This script must be run as an Administrator. Start PowerShell with the Run as Administrator option"
+        }
+
+        Write-Host -ForegroundColor Cyan "Checking the specified openHAB directory..."
+        if (!(Test-Path "$OHDirectory\userdata") -And !(Test-Path -Path "$OHDirectory\conf")) {
+            throw "$OHDirectory\userdata doesn't exist! Make sure you are in the " +
+                "openHAB directory or specify the -OHDirectory parameter!"
+        }
+
+        if ($OHDirectory -eq '.') {$OHDirectory = pwd }
+
+        if ([Environment]::GetEnvironmentVariable("OPENHAB_CONF", "Machine")) {
+            $OHConf = [Environment]::GetEnvironmentVariable("OPENHAB_CONF", "Machine")
+        } else {
+            $OHConf = "$OHDirectory\conf"
+        }
+        if ([Environment]::GetEnvironmentVariable("OPENHAB_USERDATA", "Machine")) {
+            $OHUserdata = [Environment]::GetEnvironmentVariable("OPENHAB_USERDATA", "Machine")
+        } else {
+            $OHUserdata = "$OHDirectory\userdata"
+        }
+        if (!($OHBackups)) {
+            if ([Environment]::GetEnvironmentVariable("OPENHAB_BACKUPS", "Machine")) {
+                $OHBackups = [Environment]::GetEnvironmentVariable("OPENHAB_BACKUPS", "Machine")
+            } else {
+                $OHBackups = "$OHDirectory\backups"
+                if (!(Test-Path "$OHDirectory\backups")){
+                    mkdir "$OHBackups" | Out-Null
+                }
+            }
+        }
+
+        Write-Host -ForegroundColor Yellow "Using $OHConf as conf folder"
+        Write-Host -ForegroundColor Yellow "Using $OHUserdata as userdata folder"
+        Write-Host -ForegroundColor Yellow "Using $OHBackups as backups folder"
+
+        $VersionLine = Get-Content "$OHDirectory\userdata\etc\version.properties" | Where-Object { $_.Contains("openhab-distro")}
+        $CurrentVersionIndex = $VersionLine.IndexOf(":")
+        $CurrentVersion = $VersionLine.Substring($currentVersionIndex + 2)
+        $timestamp = Get-Date -Format yyyyMMddHHmm
+        Write-Output "version=$CurrentVersion" | Set-Content "$TempDir\backup.properties"
+        Write-Output "timestamp=$timestamp" | Add-Content "$TempDir\backup.properties"
+        #Write-Output "user=$OHUser" | Add-Content "$TempDir\backup.properties"
+        #Write-Output "group=$OHGroup" | Add-Content "$TempDir\backup.properties"
+        
+        Write-Host -ForegroundColor Cyan "Copying userdata and conf folder contents to temp directory"
+        $TempDir=([Environment]::GetEnvironmentVariable("TEMP", "Machine"))+"\openhab"
```

Comment:
These lines have to be moved before L76 and L77 for the script to work properly.

Reply:
Fixed and ready to go, just waiting on your clarification above.